dell-restore-system

```diff
@@ -67,23 +69,45 @@ if __name__ == '__main__':
-                                        DBUS_INTERFACE_NAME)
-            dbus_sync_call_signal_wrapper(dbus_iface,
-                                          "enable_boot_to_restore",
-                                          {},
-                                          True)
-        except dbus.DBusException as ex:
-            if ex.get_dbus_name() == 'org.freedesktop.DBus.Error.FileNotFound':
-                text = _("Cannot connect to dbus")
-            if ex.get_dbus_name() == PermissionDeniedByPolicy._dbus_error_name:
-                text = _("Permission Denied")
+        if options.dhc_flag:
+            if check_for_restore_command():
+                try:
+                    bus = dbus.SystemBus()
+                    dbus_iface = dbus.Interface(bus.get_object(DBUS_BUS_NAME,
+                                                '/RecoveryMedia'),
+                                                DBUS_INTERFACE_NAME)
+                    dbus_sync_call_signal_wrapper(dbus_iface,
+                                                "enable_boot_to_restore_dhc",
+                                                 {},
+                                                True)
+                except dbus.DBusException as ex:
+                    if ex.get_dbus_name() == 'org.freedesktop.DBus.Error.FileNotFound':
+                        text = _("Cannot connect to dbus")
+                    if ex.get_dbus_name() == PermissionDeniedByPolicy._dbus_error_name:
+                        text = _("Permission Denied")
+                    else:
+                        text = ex.get_dbus_message()
             else:
-                text = ex.get_dbus_message()
-            sys.stderr.write("%s\n" % text)
-            sys.exit(1)
+                print("Sorry, you machine is not support the '-d' option, because your machine model is not belong to install Dell Hybrid Client series, you can try 'dell-restore-system -y' to restore os, thanks.")
+        else:
+            try:
+                bus = dbus.SystemBus()
+                dbus_iface = dbus.Interface(bus.get_object(DBUS_BUS_NAME,
+                                            '/RecoveryMedia'),
+                                            DBUS_INTERFACE_NAME)
+                dbus_sync_call_signal_wrapper(dbus_iface,
+                                              "enable_boot_to_restore",
+                                              {},
+                                              True)
+            except dbus.DBusException as ex:
+                if ex.get_dbus_name() == 'org.freedesktop.DBus.Error.FileNotFound':
+                    text = _("Cannot connect to dbus")
+                if ex.get_dbus_name() == PermissionDeniedByPolicy._dbus_error_name:
+                    text = _("Permission Denied")
+                else:
+                    text = ex.get_dbus_message()
+
+                sys.stderr.write("%s\n" % text)
+                sys.exit(1)
```

Comment:
There's a lot of code duplication between the two functions.  The only real difference should be the use of `enable_boot_to_restore_dhc` vs `enable_boot_to_restore`.  Can you please move the duplication out and just configure the variables for which interface to call and the error message handling?